olp-cpp-sdk-core/src/thread/PriorityQueueExtended.h

```diff
@@ -89,7 +89,8 @@ class PriorityQueueExtended<T, COMPARE>::PriorityQueueExtendedImpl {
     DistinguishableObject(std::uint32_t id, Args... args)
         : id(id), obj(args...) {}
     DistinguishableObject(std::uint32_t id, const T& obj) : id(id), obj(obj) {}
-    DistinguishableObject(std::uint32_t id, T&& obj) : id(id), obj(obj) {}
+    DistinguishableObject(std::uint32_t id, T&& obj)
+        : id(id), obj(std::forward<T>(obj)) {}
```

Comment:
As far as I know, this will not work as T obj is not a deducible type but a predefined type trough the  PriorityQueueExtended deduction, thus making it a rvalue by default. So theoretically the move constructor should be called when initializing the obj member with an rvalue.  So maybe the issue is somewhere else.